src/thunderbots/software/ai/world/team.cpp

```diff
@@ -1,33 +1,134 @@
 #include "team.h"
+#include <shared/constants.h>
+#include "util/parameter/dynamic_parameters.h"
 
-Team::Team()
+// We initialize the goalie_id to -1, a value that will never occur naturally as all robot
+// ids are numbers >= 0 because otherwise the goalie_id will be initialized to 0 by
+// default, and we want there to be no goalie until the value is updated.
+Team::Team() : team_robots(), goalie_id(-1)
 {
 }
 
-void Team::update(std::vector<Robot> &team_robots)
+void Team::update(const thunderbots_msgs::Team& team_msg)
```

Comment:
imo we could eliminate our code's coupling with ROS here and keep everything that touches ROS in main.cpp.

Reply:
Will look into it. Going to close this PR and break it up.